workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/EffectiveServiceSpecUtil.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.controller.directive.impl.BuildSpecJsonHandler;
+import io.hyscale.controller.directive.impl.DockerfileJsonHandler;
+import io.hyscale.controller.directive.impl.PropsJsonHandler;
+
+/**
+ * Util class to provide operations on service spec like support windows based
+ * file separator
+ * 
+ * @author tushart
+ *
+ */
+@Component
+public class EffectiveServiceSpecUtil {
+
+	private static final Logger logger = LoggerFactory.getLogger(EffectiveServiceSpecUtil.class);
+
+	@Autowired
+	private BuildSpecJsonHandler buildSpecJsonHandler;
+
+	@Autowired
+	private DockerfileJsonHandler dockerfileJsonHandler;
+
+	@Autowired
+	private PropsJsonHandler propsJsonHandler;
```

Comment:
Autowire entire list of implementation and iterate over each 
Example :  **PluginHandlers**